File: 4/calc-1.js
const fs = require("fs");
const { reduce, intersection } = require("lodash");

const input = fs.readFileSync("./input-1.txt").toString();

const assignments = input.split("\n").slice(0, -1);

const assignments2 = assignments.map((assignment) => {
  const [first, second] = assignment.split(",");
  return [first, second];
});

const assignments3 = assignments2.map(([f, s]) => {
  const [f1, f2] = f.split("-");
  const [s1, s2] = s.split("-");
  return [
    [parseInt(f1), parseInt(f2)],
    [parseInt(s1), parseInt(s2)],
  ];
});

const fullyContained = (r1, r2) => {
  if (r1[0] <= r2[0] && r1[1] >= r2[1]) {
    return true;
  }
  return false;
};

const contained = assignments3.map(([f, s]) => {
  return fullyContained(f, s) || fullyContained(s, f);
});

console.log(contained.filter(Boolean).length);

File: 4/calc-1.clj
(defn parse [xs]
  (map #(Integer/parseInt %) xs))

(defn fully-contained [r1 r2]
  (println "r1: " r1)
  (and (<= (Integer/parseInt (first r1)) (Integer/parseInt (first r2)))
       (<= (Integer/parseInt (second r1)) (Integer/parseInt (second r2)))))

(println (fully-contained [2 4] [1 3]))

(let [input (slurp "input-test.txt")]
  (as-> input v
    (str/split v #"\n") ;; split into lines
    (map (fn [a] (str/split a #",")) v) ;; split into pairs
    (map (fn [b] (map (fn [a] (str/split a #"-")) b)) v) ;; split into ranges
    (map (fn [a] (map #(fully-contained
                        (first %)
                        (second %)) a)) v)
    (println v)))

